src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -152,6 +152,10 @@ def create(self, bucket, uid, zonegroup=None, placement_target=None):
             raise DashboardException(e, http_status_code=500, component='rgw')
 
     def set(self, bucket, bucket_id, uid):
+        # When linking a non-tenant-user owned bucket to a tenanted user, we need to prefix bucket
+        # name with '/'. e.g. photos -> /photos
```

Comment:
FYI: this is because we have to use the "legacy" tenant that has an empty name
https://docs.ceph.com/docs/master/radosgw/multitenancy/
> From the Jewel release onward, each user and bucket lies under a tenant.
> For compatibility, a “legacy” tenant with an empty name is provided.